### PR TITLE
Convert ProcessTest_ConsoleApp to .NET Core

### PR DESCRIPTION
**Pull request by stephentoub**
For ProcessTest_ConsoleApp to be runnable on Unix, it needs to first be on .NET Core.

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @pallavit

**Comment by stephentoub** [comment]
Thanks, Pallavi.

File: src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs [hunk 1]
```diff
@@ -28,15 +27,17 @@ public ProcessTest()
...
-            // Also ensure that there are no open processes with the same name as ProcessName
-            foreach (Process p in Process.GetProcessesByName(ProcessName))
+            // Ensure that there are no open processes with the same name as ProcessName
+            foreach (Process p in _processes)
             {
                 if (!p.HasExited)
                 {
+                    try
+                    {
                     p.Kill();
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs]: Will fix separately as part of some more fixes coming soon. Thanks.

File: src/System.Diagnostics.Process/tests/System.Diagnostics.Process.TestConsoleApp/System.Diagnostics.Process.TestConsoleApp.csproj [hunk 2]
```diff
@@ -6,23 +6,16 @@
     <OutputType>Exe</OutputType>
     <RootNamespace>System.Diagnostics.ProcessTests</RootNamespace>
     <AssemblyName>System.Diagnostics.Process.TestConsoleApp</AssemblyName>
-    <TargetFrameworkIdentifier>.NETFramework</TargetFrameworkIdentifier>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <TargetFrameworkProfile />
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
```
Comment (weshaggard) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.TestConsoleApp/System.Diagnostics.Process.TestConsoleApp.csproj]: Is this property actually needed?

Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/System.Diagnostics.Process.TestConsoleApp/System.Diagnostics.Process.TestConsoleApp.csproj]: Without it, I get this warning:
```
warning MSB3276: Found conflicts between different versions of the same dependent assembly. Please set the "AutoGenerateBindingRedirects" property to true in the project file. For more information, see http://go.microsoft.com/fwlink/?LinkId=294190
```

**Pull request merged**